plugins/rest-api.php

```diff
@@ -0,0 +1,375 @@
+<?php
+
+/*
+ * Add support for caching REST API v2+ endpoints
+ *
+ * URLs without parameters -- e.g., https://example.org/wp-json/wp/v2/posts/ -- can be served cached files by
+ * mod_rewrite. URLs with parameters -- e.g., https://example.org/wp-json/wp/v2/posts?per_page=10 -- are served
+ * cached files by PHP.
+ *
+ */
+
+// Prevent direct access
+if ( ! defined( 'WPCACHEHOME' ) ) {
+	return;
+}
+
+/*
+ * Future development:
+ *
+ * See if there's a way to serve gzipped json while still sending the correct content-type header
+ *      can use Apache's ModMime?
+ *          Add filter to $gziprules in wpsc_get_htaccess_info
+ *          Add `<FilesMatch "\.json(\.gz)?$">` to cache/.htaccess
+ *          Add `RewriteCond %{HTTP:Accept-Encoding} gzip` rules to /.htaccess
+ *		maybe detect if server is configured to send application/json for gzipped
+ *          if it's not, then don't generate the corresponding rules?
+ *      could maybe put AddType into htaccess to set gzip mime type
+ *          thought not all servers supported, but w3 total cache does it. maybe they detect support first?
+ *
+ * When disabling the plugin, clear all cached files to get rid of the API cache, then run the preloader to regenerate everything else (if it's enabled)
+ *
+ * Need to update the cache when underlying data changes, rather than just waiting for manual expiration?
+ *      would be bad if deleted post continue to show up in cache, etc
+ *      how to tell that updating a post or category maps to an API endpoint?
+ *      worst case, could just flush all endpoint cache files when any post/comment/taxterm/etc is updated
```

Comment:
Yeah, that's a good point to bring up. I don't have any ideas for how to map content to API endpoints right now (other than just a hardcoded list for known endpoints like `/posts`). 

One thing that could help avoid scaling issues, would be to keep track of when we flush the cache, and don't do it more often then once every 5 minutes, or something like that. We'll need to do more thinking/testing/etc there, though